### PR TITLE
[G-API]: Relocation of operators' overloads

### PR DESCRIPTION
**Pull request by OrestChura**
**These changes**:
 - relocate overloaded operators for `cv::GMat` and `cv::GScalar` to `cv::` 
   namespace
 - add a test to check the compilation of operators' usage.

**Explanation**:
Overloads of math (and other) operators for `cv::GMat` and `cv::GScalar` contained in `operators.hpp` are located in a global namespace, which causes conflicts in a particular case: when an operator is called in a non-global namespace and there is another overload of this operator in this namespace - it **hides** our overload situated in the global namespace.

**Example**:
```
#include <opencv2/gapi.hpp>

namespace for_test
{
class A {};
int operator|(A, int);
int operator|(A, int) { return 1; }
cv::GComputationT<cv::GMat(cv::GMat, cv::GMat)> cmp([&](cv::GMat in1, cv::GMat in2)
{
    return in1 | in2;
});
} // namespace for_test
```

**Sources**:
 - [Namespaces and operator resolution]( https://stackoverflow.com/questions/5195512/namespaces-and-operator-resolution);
 - [Where should non-member operator overloads be placed](https://stackoverflow.com/questions/3623631/where-should-non-member-operator-overloads-be-placed)
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

```
force_builders=Custom,Custom Win,Custom Mac
build_gapi_standalone:Linux x64=ade-0.1.1f
build_gapi_standalone:Win64=ade-0.1.1f
build_gapi_standalone:Mac=ade-0.1.1f
build_gapi_standalone:Linux x64 Debug=ade-0.1.1f

Xbuild_image:Custom=centos:7
Xbuildworker:Custom=linux-1
build_gapi_standalone:Custom=ade-0.1.1f

build_image:Custom=ubuntu-openvino-2020.3.0:16.04
build_image:Custom Win=openvino-2020.3.0
build_image:Custom Mac=openvino-2020.3.0

test_modules:Custom=gapi
test_modules:Custom Win=gapi
test_modules:Custom Mac=gapi

buildworker:Custom=linux-1
# disabled due high memory usage: test_opencl:Custom=ON
test_opencl:Custom=OFF
test_bigdata:Custom=1
test_filter:Custom=*
```

### CODE REVIEW

**Comment by OrestChura** [comment]
@AsyaPronina please review

**Review by AsyaPronina — APPROVED** [review]

**Comment by OrestChura** [comment]
@anton-potapov please take a look

**Comment by anton-potapov** [comment]
@OrestChura , overall looks good , but please add tests for all operators.

File: modules/gapi/test/common/gapi_operators_tests_inl.hpp [hunk 1]
```diff
@@ -77,6 +77,103 @@ TEST_P(NotOperatorTest, OperatorAccuracyTest)
...
+
+int operator<=(Foo, int);
+int operator<=(Foo, int) { return 1; }
+
+int operator==(Foo, int);
+int operator==(Foo, int) { return 1; }
+
+int operator!=(Foo, int);
+int operator!=(Foo, int) { return 1; }
+
```
Comment (anton-potapov) [inline modules/gapi/test/common/gapi_operators_tests_inl.hpp:123]: `~` is missing

File: modules/gapi/test/common/gapi_operators_tests_inl.hpp [hunk 2]
```diff
@@ -77,6 +77,103 @@ TEST_P(NotOperatorTest, OperatorAccuracyTest)
         EXPECT_EQ(0, cvtest::norm(out_mat_ocv, out_mat_gapi, NORM_INF));
     }
 }
+
+namespace for_test
+{
+class Foo {};
+
+int operator&(Foo, int);
+int operator&(Foo, int) { return 1; }
```
Comment (anton-potapov) [inline modules/gapi/test/common/gapi_operators_tests_inl.hpp:86]: IMHO no need in these forward declarations. As well `inline`  specifier is required.

**Review by anton-potapov — CHANGES_REQUESTED** [review]

File: modules/gapi/test/common/gapi_operators_tests_inl.hpp (hunk 1 above)
Comment (OrestChura) [reply modules/gapi/test/common/gapi_operators_tests_inl.hpp:101]: Oh, it was, thank you!

added

File: modules/gapi/test/common/gapi_operators_tests_inl.hpp (hunk 2 above)
Comment (OrestChura) [reply modules/gapi/test/common/gapi_operators_tests_inl.hpp]: Yes, you're right, thanks!
reworked

**Review by anton-potapov — APPROVED** [review]

**Pull request merged**